libs/CovidDatasets.py

```diff
@@ -196,14 +196,15 @@ def combine_state_county_data(self, country, state):
             [self.DATE_FIELD, self.COUNTRY_FIELD, self.STATE_FIELD], as_index=False
         )[[self.CASE_FIELD, self.DEATH_FIELD, self.RECOVERED_FIELD]].sum()
         # Now we fill in whatever gaps we can in the state data using the county data
-        curr_date = state_data[self.DATE_FIELD].max()  # Start on the last date of state data we have
+        curr_date = max(state_data[self.DATE_FIELD].max(), county_data[self.DATE_FIELD].max()) # Start on the last date of state data we have
```

Comment:
This is an interesting thought, one I hadn't considered.

In general we try to use the state data as our first choice and fill in the holes with county data. I don't imagine that's likely for the county data goes later than the state data, but I think we should have a plan in place for when it does. 

I think that we should end when the state data ends, because we are projecting at a state level, and often times summing counties provides a lower total than the state data. In this highly hypothetical situation, it is likely that the final cycle or two of just county data would artificially drag down the model.

In other words, I don't think we should do this, just wanted to talk through why 😄

Reply:
> summing counties provides a lower total than the state data

It also sometimes yields a higher one, as is the case in Washington state.

Reply:
For JHU, though, I think each data set has exclusively been states or counties. I could be wrong, though.